metrics/web/docs/features/storybook/01_metrics_storybook_design.md

```diff
@@ -0,0 +1,199 @@
+
+Pros:
+
+- Provides an ability to manage themes.
+- Provides an ability to group widgets by their types.
+- Provides an ability to change widgets' appearance.
+
+Cons:
+
+- Provides only a plain UI.
+- Requires updating a Dart version to 2.12.0 or higher.
+- Does not have an ability to change the viewport size.
+- Does not have a docs tab that supports MDX (or similar) with the documentation about the widget.
+- Does not have a search across components functionality.
+
+4. [Storybook_flutter](https://pub.dev/packages/storybook_flutter)
+
+A cross-platform storybook for showcasing widgets. It works on all platforms supported by Flutter.
+
+Pros:
+
+- Allows managing themes.
+- Allows creating a custom Storybook (e.g., UI).
+- Allows approximating how a particular widget looks and performs on different devices (requires additional related package).
+
+Cons:
+
+- Provides only a plain UI.
+- Requires updating a Dart version to 2.12.0 or higher.
+- Does not have a docs tab that supports MDX (or similar) with the documentation about the widget.
+- Does not have a search across components functionality.
+
+#### Manually create a Storybook from scratch
+
+Another solution is to provide a completely new Flutter project with the [required functionality](#requirements).
+
+Pros:
+
+- Allows creating a custom Storybook with all required features.
+
+Cons:
+
+- Requires designing a new UI for the Metrics storybook.
+- Takes extra time to implement a new UI element.
+
+#### Decision
+
+The described above packages do not suit us for several reasons:
+
+- They are rather libraries to showcasing widgets with an ability to change their appearance, but the storybook is a bit more than that. A storybook contains documentation about widgets, information about the possible parameters they can accept, search across their functionality, and more.
+
+- The management of themes in these packages is reduced only to the possibility of switching between the dark and the light themes, as well as the presence of a toggle theme button, the behavior of which we cannot control. Even though the packages allow managing themes, it becomes a complex task to integrate the Metrics theme into the storybook. The described above packages do not allow to provide the theme data for all the Metric widgets. See [Metrics Theme guidelines](https://github.com/Flank/flank-dashboard/blob/master/metrics/web/docs/03_widget_structure_organization.md#metrics-theme-guidelines) to get more metrics theme implementation details.
+
+- The UI in these packages is, also very basic, without thoughtful design and UX.
+
+Summing up the consideration of approaches and given the above explanations, the best solution would be to create this library from [scratch](#manually-create-a-storybook-from-scratch).
+
+### Prototyping
```

Comment:
🔨 I think the prototyping section should contain information that confirms that the feature implementation is possible. For example, we can prove that we can extract the part of the Metrics Web application into a separate package by providing appropriate code snippets. What do you think?